### PR TITLE
Simplified using directive comparer

### PR DESCRIPTION
**Pull request by svick**
Simplified `UsingsAndExternAliasesDirectiveComparer` by creating a `private enum` to help with sorting the different kinds of `using`s.

This PR makes the code shorter, less repetitive and I think also easier to understand. It also remove some unnecessary `else`s.

Though this doesn't improve functionality in any way, so feel free to close this PR if you think it's not worth it.

### CODE REVIEW

**Comment by jmarolf** [comment]
@dotnet-bot retest windows_release_unit32_prtest please

**Comment by jmarolf** [comment]
@dotnet-bot retest windows_release_unit64_prtest please

File: src/Workspaces/CSharp/Portable/Utilities/UsingsAndExternAliasesDirectiveComparer.cs [hunk 1]
```diff
@@ -29,6 +29,31 @@ internal class UsingsAndExternAliasesDirectiveComparer : IComparer<SyntaxNode>
...
+        {
+            if (externDirective != null)
+            {
+                return UsingKind.Extern;
+            }
+            if (usingDirective.Alias != null)
+            {
+                return UsingKind.Alias;
+            }
+            if (usingDirective.StaticKeyword.IsKind(SyntaxKind.StaticKeyword))
```
Comment (CyrusNajmabadi) [inline src/Workspaces/CSharp/Portable/Utilities/UsingsAndExternAliasesDirectiveComparer.cs]: more idiomatic is ".StaticKeyword != default".

File: src/Workspaces/CSharp/Portable/Utilities/UsingsAndExternAliasesDirectiveComparer.cs [hunk 2]
```diff
@@ -41,64 +66,31 @@ public int Compare(SyntaxNode directive1, SyntaxNode directive2)
...
-            }
-            else if (directive2IsAlias && !directive1IsAlias)
+            if (directive1Kind > directive2Kind)
             {
                 return 1;
             }
 
             // ok, it's the same type of using now.
-            if (directive1IsExtern)
+            if (directive1Kind == UsingKind.Extern)
```
Comment (CyrusNajmabadi) [inline src/Workspaces/CSharp/Portable/Utilities/UsingsAndExternAliasesDirectiveComparer.cs]: consider switching (har har) to switch.

File: src/Workspaces/CSharp/Portable/Utilities/UsingsAndExternAliasesDirectiveComparer.cs [hunk 3]
```diff
@@ -41,64 +66,31 @@ public int Compare(SyntaxNode directive1, SyntaxNode directive2)
...
+            var directive2Kind = GetUsingKind(using2, extern2);
 
             // different types of usings get broken up into groups.
             //  * externs
             //  * usings
             //  * using statics
             //  * aliases
 
-            if (directive1IsExtern && !directive2IsExtern)
+            if (directive1Kind < directive2Kind)
```
Comment (CyrusNajmabadi) [inline src/Workspaces/CSharp/Portable/Utilities/UsingsAndExternAliasesDirectiveComparer.cs]: can simplify by subtracting, then just returning the value if !=0.

**Comment by svick** [comment]
@CyrusNajmabadi Done.

File: src/Workspaces/CSharp/Portable/Utilities/UsingsAndExternAliasesDirectiveComparer.cs [hunk 4]
```diff
@@ -41,81 +66,41 @@ public int Compare(SyntaxNode directive1, SyntaxNode directive2)
...
-
-            var directive1IsNamespace = using1 != null && using1.Alias == null && !using1.StaticKeyword.IsKind(SyntaxKind.StaticKeyword);
-            var directive2IsNamespace = using2 != null && using2.Alias == null && !using2.StaticKeyword.IsKind(SyntaxKind.StaticKeyword);
-
-            var directive1IsUsingStatic = using1 != null && using1.StaticKeyword.IsKind(SyntaxKind.StaticKeyword);
-            var directive2IsUsingStatic = using2 != null && using2.StaticKeyword.IsKind(SyntaxKind.StaticKeyword);
-
-            var directive1IsAlias = using1 != null && using1.Alias != null;
-            var directive2IsAlias = using2 != null && using2.Alias != null;
+            UsingKind directive1Kind = GetUsingKind(using1, extern1);
```
Comment (CyrusNajmabadi) [inline src/Workspaces/CSharp/Portable/Utilities/UsingsAndExternAliasesDirectiveComparer.cs]: nit: use var

**Comment by sharwell** [comment]
@Pilchie What milestone should this clean up be?

**Comment by Pilchie** [comment]
I would leave it in master unless there is a customer scenario that it improves, or we believe it's likely to conflict heavily with other planned work in this area in a  sooner milestone.

**Comment by sharwell** [comment]
@dotnet-bot retest this please

**Pull request merged**